src/toxcore/onion/packet/onion_announce_response.rs

```diff
@@ -142,7 +139,8 @@ impl FromBytes for OnionAnnounceResponsePayload {
         announce_status: call!(AnnounceStatus::from_bytes) >>
         ping_id_or_pk: call!(sha256::Digest::from_bytes) >>
         nodes: many0!(PackedNode::from_bytes) >>
-        cond_reduce!(nodes.len() <= 4, eof!()) >>
+        _len: verify!(value!(nodes.len()), |len| *len <= 4 as usize) >>
```

Comment:
You don't need to create an unused variable `_len`, you may use `verify!` without `_len: ` IIRC.